packages/components/text/src/Text.tsx

```diff
@@ -119,14 +125,30 @@ export const Text = compressible<TextProps, TextTokens>((props: TextProps, useTo
 
   // return a continuation function that allows this text to be compressed
   return (extra: TextProps, children: React.ReactNode) => {
+    // [TODO(#2268): Remove once RN Core properly supports Dynamic Type scaling
+    interface ScaleStyleAdjustmentProps {
+      fontSize?: number;
+      lineHeight?: number;
+    }
+    let scaleStyleAdjustmentProps: ScaleStyleAdjustmentProps = {};
+    if (dynamicTypeVariant !== undefined && typeof tokenStyle.fontSize === 'number' && typeof tokenStyle.lineHeight === 'number') {
+      const scaleFactor = fontMetricsScaleFactors[dynamicTypeVariant] ?? 1;
+      scaleStyleAdjustmentProps = {
+        fontSize: tokenStyle.fontSize * scaleFactor,
+        lineHeight: tokenStyle.lineHeight * scaleFactor,
+      };
+    }
```

Comment:
Is there any way we can simplify this if () calculation? or move it to a separate function / into the existing hook?

Reply:
The problem is that `tokenStyle.fontSize` can also be a `string` (e.g., `"14px"`). Ditto for `tokenStyle.lineHeight`. We could try to parse out the numerical value, but that would make things more complicated.

It works out because all of our font size tokens are pure numbers, but I don't know if this is an assumption we absolutely want to stick with. Maybe we should clarify this?

Reply:
Talked offline about this. I feel like this makes a good case that we should enforce a number as much as possible. If we can do that in our tokens, great. If not, let's at least add a comment on top of these checks?

Reply:
Done.